### ISSUE TITLE
Need to fix strand issue in "segment" module

### ISSUE THREAD

**Issue by magruca** [Dowell-Lab/DAStk#2]
In GitLab by @magr0763 on Sep 28, 2018, 09:31

This issue is similar to the strand issue in the "train" module already listed. Here again, in the documentation you can specify "--strand both", however the argument in the code is "--strand ." in main_segment.cpp (lines 54, 91&101 **commented out**, 108, 131 **commented out**). 

In this case, however, it would be useful if you could provide segment a concatenated pos/neg bedGraph and have it output information into one bed file for both pos/neg strands. In other words, it would run as it would if strand==+ and strand==-, yet concatenate the output. This would be nice, but not entirely necessary. What would be nice if this cannot get done in time is to be able to input the concatenated bedGraph, and have it only train on +/- based on strand specification, and auto-detect based on integer sign in the 4th column as to which features to model (i.e. remove all negated values if you want to train on the pos strand only). This would be a more hacky fix, but probably easier. The easiest alternative to both of these options is we simply specify in the user guide that the user MUST provide a strand specific file, not a concatenated file. This is a little annoying as the user will then likely want to concatenate the outputs.

My other thought on this is we could easily add a 5th column to the file to annotate +/- strand if that would be easier in terms of separating lines in code. 

In summary, it would be nice if file format provided did not matter (pos only, neg only, concatenated) as this should not ultimately change performance. That said, FStitch can function as is, so a simple patch in the meantime in terms of user specs could be sufficient.

**Comment by magruca** [comment]
In GitLab by @magr0763 on Sep 28, 2018, 09:45

changed the description

**Comment by magruca** [comment]
In GitLab by @magr0763 on Oct 26, 2018, 13:27

It looks like this already exists in the module, but the variables are not defined appropriately in the ParamWrapper/main_segment.cpp to take advantage of this code (see Issue 10 ).

That said, I have now made a couple of bug fixes and updated the documentation such that you will not receive an error message and you can provide either a pos/neg single-stranded or concatenated .bedGraph file as long as the user specifies the --strand argument. This is also reflected in the help documentation.

It seems there is also a built-in error check to see if the training data was generated for pos or neg strand (see Issue #9 ). Given the time, I will explore this more.

That said, this issue is more or less resolved as stands, so I'm going to close it.

**Comment by magruca** [comment]
In GitLab by @magr0763 on Oct 26, 2018, 13:27

closed
